doc/architectural/compilation-and-development.md

```diff
@@ -194,8 +194,10 @@ tools. All of these could be improved and patches are very welcome. In
 some cases the algorithms used are described in the relevant papers.
 
 The doxygen documentation can be [accessed online](http://cprover.diffblue.com).
-To build it locally, run `doxygen` in `src/`. HTML output will be created in
-`doc/html/`. The index page is `doc/html/index.html`.
+To build it locally, run `scripts/run_doxygen.sh`. (This script will filter out
+expected warning messages from doxygen, so that new problems are more obvious.)
+HTML output will be created in `doc/html/`. The index page is
+`doc/html/index.html`.
```

Comment:
Since this PR introduces the reference file scripts/expected_doxygen_warnings.txt that will need to be maintained (e.g. removing expected warnings if they get fixed, or adding new expected warnings where we introduce stuff that is correct, but doxygen warns about due to its foibles) could you add a little more to the docs here to explain the appropriate actions?

Reply:
Done.